omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java

```diff
@@ -62,8 +65,12 @@ protected void initialize() {
 		setDefaultResponseEncoding(EncodingEnum.JSON);
 		registerInterceptor(loggingInterceptor);
 		
-		getFhirContext().setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(
-		        FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		List<String> propertyFiles = new ArrayList<>(
+		        Arrays.asList(globalPropertyService.getGlobalProperty(FhirConstants.NARRATIVES_OVERRIDE_PROPERTY_FILE, ""),
+		            FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		propertyFiles.removeAll(Arrays.asList("", null));
```

Comment:
Sometimes, it's better to do things the more straight-forward way.

```suggestion
		String narrativeOverridePropertyFile = globalPropertyService.getGlobalProperty(FhirConstants.NARRATIVES_OVERRIDE_PROPERTY_FILE);
		
		String[] narrativePropertiesFiles;
		if (narrativeOverridePropertyFile != null) {
		    narrativePropertiesFiles = new String[] { FhirConstants.HAPI_NARRATIVE_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE, narrativeOverridePropertyFile };
		} else {
			narrativePropertiesFiles = new String[] { FhirConstants.HAPI_NARRATIVE_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE }
		}
```

This results in fewer throw-away objects.

Reply:
Also, I'm unclear on this, but is the order in which properties files are loaded significant (I would guess so) and if so, which order is correct? Naïvely, I would think that you want the base file specified before the override file(s), but I haven't checked the implementation.

Reply:
I actually did it this way initially, but then changed it to maintain coverage.

Reply:
> Also, I'm unclear on this, but is the order in which properties files are loaded significant (I would guess so) and if so, which order is correct?

The override file has to be specified first to give it more priority. Yeah, it is kind of counter-intuitive, but this is the order it follows.